packages/app/src/server/routes/api/cortex/index.ts

```diff
@@ -31,6 +32,28 @@ export const cortexProxy = httpProxy.createProxyServer({ ignorePath: true });
  */
 cortexProxy.on("proxyReq", onProxyReq);
 
+cortexProxy.on("proxyRes", (proxyRes, req, res) => {
+  let data = "";
+  proxyRes.on("data", function (chunk) {
+    data += chunk;
+  });
+  proxyRes.on("end", function () {
+    if (proxyRes.headers["content-type"]?.includes("text/html")) {
+      /* some error messages are returned via HTML and status code 200 OK.
+       * We have mainly seen it with configuration errors,
+       * e.g. sharding is disabled. Hence we set the status code 
+       * to 412 Precondition Failed.
+       * 
+       * For example HTML responses please check the tests.
+       */
```

Comment:
Cortex might return other errors via HTML, which would not fall into the category of misconfiguration. But I do not know.